shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1556,12 +1604,23 @@ - (void)setupKeyboardAnimationVsyncClient {
       // Ensure the keyboardAnimationView is in view hierarchy when animation running.
       [flutterViewController.get().view addSubview:[flutterViewController keyboardAnimationView]];
     }
-    if ([flutterViewController keyboardAnimationView].layer.presentationLayer) {
-      CGFloat value =
-          [flutterViewController keyboardAnimationView].layer.presentationLayer.frame.origin.y;
-      flutterViewController.get()->_viewportMetrics.physical_view_inset_bottom = value;
-      [flutterViewController updateViewportMetrics];
+
+    CGFloat newY;
+    if ([flutterViewController keyboardSpringCurve] == nil) {
+      newY =
+          flutterViewController.get().keyboardAnimationView.layer.presentationLayer.frame.origin.y;
+    } else {
+      double start = flutterViewController.get().originalViewInsetBottom;
+      double end = flutterViewController.get().targetViewInsetBottom;
+      fml::TimeDelta timeElapsed = recorder.get()->GetVsyncTargetTime() -
+                                   flutterViewController.get().keyboardAnimationStartTime;
+      double keyboardAnimationStop =
+          [[flutterViewController keyboardSpringCurve] curveFunc:timeElapsed.ToSecondsF()];
+      newY = start + (end - start) * keyboardAnimationStop;
     }
+
+    flutterViewController.get()->_viewportMetrics.physical_view_inset_bottom = newY;
+    [flutterViewController updateViewportMetrics];
```

Comment:
Maybe it should be like this below ? , if the `presentationLayer` is nil , we should not update the viewport. The the same like old one.
**I think after making this change the `Mac Unopt Test` will passed. ^_^**
```objc
    if ([flutterViewController keyboardSpringCurve] == nil) {
      if (flutterViewController.get().keyboardAnimationView.layer.presentationLayer) {
        flutterViewController.get()->_viewportMetrics.physical_view_inset_bottom =
            flutterViewController.get()
                .keyboardAnimationView.layer.presentationLayer.frame.origin.y;
        [flutterViewController updateViewportMetrics];
      }
    } else {
      double start = flutterViewController.get().originalViewInsetBottom;
      double end = flutterViewController.get().targetViewInsetBottom;
      fml::TimeDelta timeElapsed = recorder.get()->GetVsyncTargetTime() -
                                   flutterViewController.get().keyboardAnimationStartTime;
      double keyboardAnimationStop =
          [[flutterViewController keyboardSpringCurve] curveFunc:timeElapsed.ToSecondsF()];
      flutterViewController.get()->_viewportMetrics.physical_view_inset_bottom =
          start + (end - start) * keyboardAnimationStop;
      [flutterViewController updateViewportMetrics];
    }
```

Reply:
just pushed, hopefully it passes now :)